ivy_tests/test_ivy/helpers/function_testing.py

```diff
@@ -647,64 +650,107 @@ def test_frontend_function(
         importlib.import_module("ivy.functional.frontends.jax").config.update(
             "jax_enable_x64", True
         )
-    ret = get_frontend_ret(frontend_fn, *args_ivy, **kwargs_ivy)
+
```

Comment:
We can move the frontend generation logic to the method `apply_flags` on the decorator, this removes the overhead of branching, and also makes everything cleaner and separated.

Reply:
I thought about it and I feel as though we have to keep the logic here for two reasons.

Firstly, *to_ivy is called on the args in the function body after `create_args_kwargs` which would spoil initial creation of frontend arrays in apply flags.

Secondly, although a unified variable to frontend array creation is created, it would still require us to know for which frontend do we want the create the arrays for.. and apply flags or the Flags class doesn't hv any parameters for passing in the current frontend (for which the test is being run). We also cannot work with just the np frontend arrays the way we can in the backends... as the transition between frontend arrays is not smooth (at least now) as opposed to the backends.

I hope that makes sense. Feel free to counter...